huaweicloud/services/live/resource_huaweicloud_live_url_validation.go

```diff
@@ -0,0 +1,248 @@
+	)
+
+	client, err := cfg.NewServiceClient("live", region)
+	if err != nil {
+		return diag.Errorf("error creating Live client: %s", err)
+	}
+
+	getPath := client.Endpoint + getHttpUrl
+	getPath = strings.ReplaceAll(getPath, "{project_id}", client.ProjectID)
+	getPath = fmt.Sprintf("%s?domain=%v", getPath, domainName)
+
+	getOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+
+	getResp, err := client.Request("GET", getPath, &getOpt)
+	if err != nil {
+		return common.CheckDeletedDiag(d, common.ConvertExpected400ErrInto404Err(err, "error_code", domainNameNotExistsCode),
+			"error retrieving URL validation")
+	}
+
+	getRespBody, err := utils.FlattenResponse(getResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	key := utils.PathSearch("key", getRespBody, "").(string)
+	if key == "" {
+		return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{}, "URL validation")
+	}
+
+	mErr := multierror.Append(
+		d.Set("region", region),
+		d.Set("domain_name", domainName),
+		d.Set("key", utils.PathSearch("key", getRespBody, nil)),
+		d.Set("auth_type", utils.PathSearch("auth_type", getRespBody, nil)),
+		d.Set("timeout", utils.PathSearch("timeout", getRespBody, nil)),
+	)
+
+	if err = mErr.ErrorOrNil(); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+func resourceUrlValidationUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg    = meta.(*config.Config)
+		region = cfg.GetRegion(d)
+	)
+
+	client, err := cfg.NewServiceClient("live", region)
+	if err != nil {
+		return diag.Errorf("error creating Live client: %s", err)
+	}
+
+	if d.HasChanges("key", "auth_type", "timeout") {
+		updateHttpUrl := "v1/{project_id}/guard/key-chain"
+		updatePath := client.Endpoint + updateHttpUrl
+		updatePath = strings.ReplaceAll(updatePath, "{project_id}", client.ProjectID)
+		updatePath = fmt.Sprintf("%s?domain=%v", updatePath, d.Get("domain_name"))
+		updateOpt := golangsdk.RequestOpts{
+			KeepResponseBody: true,
+			JSONBody:         utils.RemoveNil(buildUrlValidationBodyParams(d)),
+		}
+
+		_, err := client.Request("PUT", updatePath, &updateOpt)
+		if err != nil {
+			return diag.Errorf("error updating URL validation: %s", err)
+		}
+	}
+
+	return resourceUrlValidationRead(ctx, d, meta)
+}
+
+func resourceUrlValidationDelete(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg           = meta.(*config.Config)
+		region        = cfg.GetRegion(d)
+		deleteHttpUrl = "v1/{project_id}/guard/key-chain"
+	)
+
+	client, err := cfg.NewServiceClient("live", region)
+	if err != nil {
+		return diag.Errorf("error creating Live client: %s", err)
+	}
+
+	deletePath := client.Endpoint + deleteHttpUrl
+	deletePath = strings.ReplaceAll(deletePath, "{project_id}", client.ProjectID)
+	deletePath = fmt.Sprintf("%s?domain=%v", deletePath, d.Get("domain_name"))
+	deleteOpts := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
```

Comment:
The DELETE method supports checking the 204 status code by default, so no special instructions are needed here.